treee111/wahooMapsCreator
# [FEAT] Add logging of timing for total time, each main section and for each tile generation step

## This PR…

Add logging of duration of various steps of the process, to be able to see where the focus for improvements should be, and to be able to detect any changes in performance from version to version, or from machine to machine, or from country to country.

## Considerations and implementations

There are a couple of commits at the start of the history which are really general small fixups, I could have created a separate PR for them.

Ended up adding a Timings class for handling the actual duration and generating the text for logging, or doing the actual logging, based on PR comments from @treee111 .

I have just added simple logging, and have not looked for any general "metrics" python library.

Only wall time is logged.
Originally I also included CPU time in most places, but I think the CPU time it less interesting, since it is mainly other processes that are being spawned.
I have not added an CLI option to choose whether you want the logging or not, I assume it is quite interesting for most people.

I have added debug level logging for each tile, to get the duration taken for each tile.

Originally logged the time with 5 decimals, but reduced it to 3, even though that means some steps are shown as taking 0.000 seconds.

I have made one final restructuring of the commits, have kept the main parts of the commit from @treee111 for the PR review, a couple of minor suggestions there I splitted out, and combined with my commits, to have a cleaner commit history, with fewer "correction" commits.

## How to test

1. Run the CLI to generate a tile or a country, and observe the logging of duration. e.g. `python -m wahoomc cli -co malta`
2. Run the CLI to generate a tile or a country, and observe the logging of duration. e.g. `python -m wahoomc cli -co malta -v`
3. Run the unit tests with "python -m unittest -v" to see the logging.


## Pull Request Checklist
    + Especially the [Pull-Requests](https://github.com/treee111/wahooMapsCreator/blob/develop/docs/CONTRIBUTING.md#Pull-Requests) section
